### PR TITLE
Update Mono to 5.18.0.240 (release)

### PR DESCRIPTION
**Pull request by m4tt075**
### Motivation
Memory leakage with `Mono 5.8.0` and breaking of recent `Jackett` packages.
Linked issue: #3470 

### Notes
- Updates `Mono` to 5.18.0.240 (release)
- Introduces new patch and `autogen.sh` to circumvent build regression with recent mono builds (c.f. [Mono issue 9951](https://github.com/mono/mono/issues/9951))
- Supports PowerPC (qoriq) platform from DSM6 onwards
- Test 6.1 and 6.2 packages available under #3470. Please also report test results there. 
- WIP as this needs to be tested in conjunction with depending packages (`Jackett`, `Sonarr`, etc)

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [ ] New installation of package completed successfully

### CODE REVIEW

**Comment by ymartin59** [comment]
I also wonder if recommendations for embedded systems may be apply too: https://www.mono-project.com/docs/compiling-mono/small-footprint/

**Comment by m4tt075** [comment]
@ymartin59 
Re ppc: Had the same thought. :-)  Tried it, but continued to run into problems. Added one or two patches and still could not get it to work. Gave up after about 3 hours. 
Re embedded: Have not seen this before. I believe we strip all binaries by default already, but will look into the other recos...

**Comment by m4tt075** [comment]
Just pushed updates to latest release (5.18.0.240). Also included a build configuration to optimize for size.

File: spk/mono/Makefile [hunk 1]
```diff
@@ -36,3 +36,5 @@ mono_extra_install:
 mono_pre_copy: $(INSTALL_PLIST)
 	# remove boehm GC references from PLIST if it isn't available on the platform (e.g. rtd1296)
```
Comment (ymartin59) [inline spk/mono/Makefile]: @m4tt075 Would it be possible to use recent per-cross PLIST_TRANSFORM with per-arch condition? I guess not only rtd1296 is concerned by "mono-boehm" but all ARM8 so aarch64 virtual arch.

**Comment by m4tt075** [comment]
@ymartin59 Sigh. I have docker build environment issues again...  :-(
All (at least ARM-based) docker builds fail with:
```
./autogen.sh: line 155: ./autogen.sh: No such file or directory
configure: error: in `/spksrc/spk/mono/work-alpine-6.1/mono-5.18.0.240':
configure: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
```
See https://travis-ci.org/m4tt075/spksrc/jobs/498936598 for details. The same `alpine` build works fine in my Virtualbox environment. Any idea what cause this issue after the framework changes or what I have to change?

**Comment by m4tt075** [comment]
Hey, `qoriq` compiles now...  :-)
Looking good, but need more testing. For another day...

**Comment by m4tt075** [comment]
Just force-pushed code cleanup (including `PLIST_TRANSFORM` usage) and build improvements which should (FINALLY!) allow `mono` use on `qoriq` platforms.

**Comment by m4tt075** [comment]
@ymartin59 I've successfully cross-compiled `Jackett`, `Lidarr`, `Radarr` and `Sonarr` for `qoriq-6.1` with the additional commits I've just pushed. Test packages here:
https://github.com/m4tt075/spksrc/releases/tag/Mono-qoriq-deps
Let's wait for feedback before we merge and publish. I'll watch the download numbers and come back.

**Comment by m4tt075** [comment]
OK, `qoriq` support dropped again. See #3470 for details and rationale.
Non-ppc platforms are still being tested. 12 downloads of non-qoriq platforms so far. Keeping this open as WIP until more (positive) feedback is available.

**Comment by matej** [comment]
Sorry to resurrect this PR, but is there any news on the status of those package updates? I'm seeing mono taking up pretty significant CPU resources on my system and am wondering if updating could improve the situation.

**Comment by m4tt075** [comment]
Yes, they have been published. There are (upstream) issues with `aarch64` platforms (see #3470), but especially `x64` platforms should work fine with the update. If you have additional questions, please ask them in #3470. PRs are for developers to discuss code-related questions.

**Comment by naikdipe** [comment]
> ### Motivation
> Memory leakage with `Mono 5.8.0` and breaking of recent `Jackett` packages.
> Linked issue: #3470
> 
> ### Notes
> * Updates `Mono` to 5.18.0.240 (release)
> * Introduces new patch and `autogen.sh` to circumvent build regression with recent mono builds (c.f. [Mono issue 9951](https://github.com/mono/mono/issues/9951))
> * Supports PowerPC (qoriq) platform from DSM6 onwards
> * Test 6.1 and 6.2 packages available under #3470. Please also report test results there.
> * WIP as this needs to be tested in conjunction with depending packages (`Jackett`, `Sonarr`, etc)
> 
> ### Checklist
> * [x]  Build rule `all-supported` completed successfully
> * [x]  Package upgrade completed successfully
> * [ ]  New installation of package completed successfully

Hello, is the QORIQ download still available? I do not see it here:

https://github.com/m4tt075/spksrc/releases/

**Comment by naikdipe** [comment]
Apologies, I just saw this 
"OK, qoriq support dropped again. See #3470 for details and rationale."

**Pull request merged**